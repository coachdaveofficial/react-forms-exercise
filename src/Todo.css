.Todo {
    color: black;
    font-weight: bold;
    margin: 20px;
    padding: 5px;
}

.Completed {
    color: gray;
    font-weight: 200;
    text-decoration: line-through;
}