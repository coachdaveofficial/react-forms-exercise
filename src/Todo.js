import React from 'react';
import './Todo.css'

function Todo({isCompleted=false, task, remove, id, complete}) {

    const classes = `Todo ${isCompleted ? "Completed" : ""}`
    return (
        <li data-testid="Todo" id={id} className={classes}>
            {task}
            <span style={{'marginLeft': '10px'}}>
                <button onClick={() => remove(id)}>Remove me</button>
                <button onClick={() => complete(id)}>Completed</button>
            </span>
        </li>
    )
}



export default Todo;