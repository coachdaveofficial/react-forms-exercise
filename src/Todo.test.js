import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function () {
    render(<Todo />)
});

it("matches snapshot", function () {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders with the provided task', () => {
    const { getByTestId } = render(<Todo task="test" />);

    const todoElement = getByTestId('Todo');

    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('test');
});

it('calls remove function with todo ID when "Remove me" button is clicked', () => {
    const removeMock = jest.fn();
    const todoId = 'test';
    const { getByText } = render(<Todo task="test" remove={removeMock} id={todoId} />);

    const removeButton = getByText('Remove me');

    fireEvent.click(removeButton);

    expect(removeMock).toHaveBeenCalledWith(todoId);
});