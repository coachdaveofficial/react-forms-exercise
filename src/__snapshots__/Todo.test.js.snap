// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo "
    data-testid="Todo"
  >
    <span
      style="margin-left: 10px;"
    >
      <button>
        Remove me
      </button>
      <button>
        Completed
      </button>
    </span>
  </li>
</DocumentFragment>
`;
